doc/stdlib/arrays.rst

```diff
@@ -176,12 +240,48 @@ Constructors
 
    n-by-n identity matrix
 
+   ::
+              eye(m, n)
+
+   m-by-n identity matrix
+
+   ::
+              eye(A)
+
+   Constructs an identity matrix of the same dimensions and type as ``A``.
+
 .. function:: eye(m, n)
 
+   ::
+              eye(n)
+
+   n-by-n identity matrix
+
+   ::
+              eye(m, n)
+
    m-by-n identity matrix
 
+   ::
+              eye(A)
+
+   Constructs an identity matrix of the same dimensions and type as ``A``.
```

Comment:
Yes, this is the methods issue mentioned in the PR description – three methods are listed in the manual, the methods are combined for the whole function, each method in the manual gets the docs for the whole function.